### ISSUE TITLE
Training Config for lusxvr/nanoVLM-230M-8k

### ISSUE THREAD

**Issue by aidansmyth95** [huggingface/nanoVLM#198]
Hi @lusxvr ,

I am looking forward to using your excellent FineVision dataset on HuggingFace and have some questions.

According to [this article](https://huggingface.co/spaces/HuggingFaceM4/FineVision), I believe your HuggingFace models (such as lusxvr/nanoVLM-230M-8k) were trained on the FineVision dataset using this repo (latest version) for 20k steps on 32 H100 GPUs. Is that correct?

I would really appreicate if you could share:
- the model configs for each of the 4 nanoVLM models you have on HuggingFace.
- the git commit of this repo that you trained those models using in case the latest commit does not perform the same
- Did you modify HuggingFaceM4/FineVision or train on it as-is? The reason I ask is that in the model config in this repo, HuggingFaceM4/FineVision_concat_shuffled_2 is referenced, which I presume is the same thing?

I am trying to replicate the better response behaviour of the models on your HuggingFace, since the models I trained with v0.1 of this repo tend to ramble a bit and hallucinate and your models do not.

Links I found useful for this issue ticket:
- https://huggingface.co/lusxvr/nanoVLM-230M-8k
- https://huggingface.co/datasets/HuggingFaceM4/FineVision
- https://huggingface.co/spaces/HuggingFaceM4/FineVision

**Comment by eva10084** [comment]
你好，方便问一下您的实验环境嘛，我的不同版本的包之间存在冲突

**Comment by aidansmyth95** [comment]
@andimarafioti or @ariG23498 - would you know the answer to this? Seems a major blocker for most of us using this repo.

**Comment by lusxvr** [comment]
Hi, 

the model config is available on the model pages itself (e.g. https://huggingface.co/lusxvr/nanoVLM-230M-8k/blob/main/config.json).

We trained all the models end of August / beginning of September, this commit should be representative (https://github.com/huggingface/nanoVLM/tree/9de5e17ac2f4c578c32085131d966464cdd252b5).

For the trained models we didn't modify FineVision, but afterwards Andi made some improvements to the dataset streaming and created a pre-shuffled version for ease of use.

**Comment by aidansmyth95** [comment]
@lusxvr Thank you very much for the commit link. I will use this info to try train it ourselves. Really appreciate your help with this, thank you again!

I presume that the train_dataset_path below is just FineVision. Will update this thread with benchmark and generation results once training concludes.

```python

@dataclass
class TrainConfig:
    lr_mp: float = 0.00512
    lr_vision_backbone: float = 5e-5 #0.0005 #
    lr_language_backbone: float = 5e-5 #0
    data_cutoff_idx: int = None
    val_ratio: float = 0.005
    batch_size: int = 1
    gradient_accumulation_steps: int = 8
    max_grad_norm: float = 1.0
    eval_in_epochs: bool = True
    eval_interval: int = 500
    stats_log_interval: int = 100
    max_training_steps: int = 80100
    max_images_per_example: int = 8
    max_images_per_knapsack: int = 36
    max_sample_length: int = 8192
    compile: bool = False
    resume_from_vlm_checkpoint: bool = False # Indicate if the training should be resumed from a checkpoint of the whole VLM or you want to start from scratch
    train_dataset_path: str = '/fsx/luis_wiedmann/datasets/asterix_rated'
    train_dataset_name: tuple[str, ...] = ("all", ) #('allava_laion', 'allava_vflan', 'cambrian(filtered)_processed', 'LLaVA_Instruct_150K', 'mmevol', 'sharegpt4o', 'sharegpt4v(coco)', 'sharegpt4v(knowledge)', 'sharegpt4v(llava)', 'sharegpt4v(sam)') # 'vision_flan(filtered)', 'lvis_instruct4v',
    relevance_min_rating: int = 1
    image_correspondence_min_rating: int = 1
    visual_dependency_min_rating: int = 1
    formatting_min_rating: int = 1
    wandb_entity: str = "HuggingFace" # Indicate the entity to log to in wandb
    log_wandb: bool = True
    use_lmms_eval: bool = True # Use lmms-eval for evaluation
    lmms_eval_tasks: str = 'mmstar,mmmu,ocrbench,textvqa,docvqa,scienceqa,mme,infovqa' # Pass additional task as one string, seperated by commas without spaces (e.g. 'mmstar,mmmu,ocrbench')
    lmms_eval_limit: float = None
    lmms_eval_batch_size: int = 64
```

**Comment by lusxvr** [comment]
> I presume that the train_dataset_path below is just FineVision.

Yes, this was just our internal working name

**Comment by aidansmyth95** [comment]
I see the need to concatenate and shuffle now. Some of the datasets are small (a few parquet files), and in our multi-GPU training code, we expect at least one parquet file per dataset per GPU, leading to indexing errors when loading datasets.

Is this the motivation for the concatenate and shuffle? If so, we should probably update the README at least to reflect this need. When I finish this work, I can push a PR with more info on that if this is the case.

**Comment by aidansmyth95** [comment]
@lusxvr would you happen to have the wandb plots for the model when you trained it, so we can compare? It will help us early on before we spend so much on compue figure out if we are trending the right direction as training proceeds or not.

**Comment by lusxvr** [comment]
I don't have access to that anymore unfortunately, but maybe @andimarafioti can help you with this.

**Comment by aidansmyth95** [comment]
Thank you @lusxvr , and thanks in advance @andimarafioti 🤞

Love FineVision, it's such a leap forward from the Cauldron.

**Comment by aidansmyth95** [comment]
@andimarafioti kindly replied to my tweet with the curves. Thank you all for helping.

<img width="2377" height="359" alt="Image" src="https://github.com/user-attachments/assets/84845779-b698-47a4-92eb-fc7a85baeb59" />

<img width="2377" height="342" alt="Image" src="https://github.com/user-attachments/assets/1212718e-9ec8-4618-9caf-c34ad416d64b" />

I will update this thread again on the recreation of those curves in the near future.

**Comment by patrickamadeus** [comment]
Hey guys, I wanna ask based on this training config [here](https://github.com/huggingface/nanoVLM/blob/9de5e17ac2f4c578c32085131d966464cdd252b5/models/config.py), how many and what GPU variant was used to train the model?

I am confused whether it was 32xH100 or not, because the number of steps shown in above graphs differ with the main README, so what was the actual effective batch size after accum + DDP/etc. ?

**Comment by aidansmyth95** [comment]
Hey Patrick, I don't think that config is representative of the pretrained models they have on HuggingFace. I think the config you point towards is for the "get training quick" example.
